runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -2006,8 +2007,7 @@ private void encodeCrc()
                     crc.update(encodeSlotByteBuffer);
 
                     long checksum = crc.getValue();
-
-                    if (crcDataLimit == recordBatch.limit())
+                    if (crcDataLimit == recordHeaderLimit && valueCompleteSize > 0)
```

Comment:
I realized that `valueCompleteSize` is always `>= 0` so this should actually be
```suggestion
                    if (crcDataLimit == recordHeaderLimit && valueCompleteSize != 0)
```
due to
```
        final int valueSize = payload != null ? payload.sizeof() : 0;
```
in `flushRecordInit`.